src/Hazelcast.Net/Clustering/MemberConnection.cs

```diff
@@ -142,8 +145,14 @@ public Action<ClientMessage> ReceivedEvent
 
         /// <summary>
         /// Gets the network address the client is connected to.
+        /// <para>In case of unisocket mode, gets the <see cref="RemoteAddress"/></para>
+        /// </summary>
+        public NetworkAddress Address { get { return _networkingOptions.SmartRouting || _networkingOptions.Cloud.Enabled ? _address : RemoteAddress; } }
```

Comment:
When `SmartRouting=false`

**Case 1: Connect to OnPrem Cluster** 

If the configured address is different than that returned by member which is connected and authenticated on(result of authention), during very first connection, client fails because of difference between addresses. Here https://github.com/hazelcast/hazelcast-csharp-client/blob/bcd328c0659c3a5b9c9edddf0cb20fc0fb64eda7/src/Hazelcast.Net/Clustering/ClusterMembers.cs#L453 returns false, due to differences and client cannot change its state to `Connected`. 
Also, that connection is removed at 
https://github.com/hazelcast/hazelcast-csharp-client/blob/bcd328c0659c3a5b9c9edddf0cb20fc0fb64eda7/src/Hazelcast.Net/Clustering/ClusterMembers.cs#L439

**Case 2: Connect to Cloud**

Member address and provided by cloud service is public and there is no need to use address provided by member during authentication which is private. 

The remote address is taken from authentication result although socket connection can be opened via configured address member replies from different address that confuses the client with current architecture. The fix does not change socket address but adds a "alias" on it with replied one by member (Go and java also behaves similar). When smart routing is enabled, situation is managed by client since it can reconnect by using member table.

Fix bypasses the https://github.com/hazelcast/hazelcast-csharp-client/blob/bcd328c0659c3a5b9c9edddf0cb20fc0fb64eda7/src/Hazelcast.Net/Clustering/ClusterMembers.cs#L453


**If configured and returned address are same, client can connect without any trouble.**